### ISSUE TITLE
Package dependencies and duplicate generation

### ISSUE THREAD

**Issue by PeterJohnson** [HebiRobotics/QuickBuffers#62]
I have A and B java packages. B depends on A (eg uses classes defined in A). Both packages also have protobuf definitions, and some of B’s .proto files use objects defined in A’s .proto files.  When I generate A, I get implementations generated for A’s protobufs, but when I generate B, I get implementations generated for *both* B’s and A’s protobufs.  This is problematic as I have other packages that pull in both A and B.

How do I prevent QuickBuffers from generating duplicate implementations of A’s protobufs in B?  Is there a way to suppress generation of A’s protobufs when building B?

**Comment by ennerf** [comment]
It's not a use case I've run into before, but I think the easiest way would be to run a secondary step that deletes the duplicates classes from B. How does Google's implementation handle this case?

**Comment by PeterJohnson** [comment]
For C++, if you run e.g. `protoc -Iwpimath/src/main/proto --cpp_out=tmp wpimath/src/main/proto/photon_types.proto`, it only generates a photon_types.pb.cc/.h.  --quickbuf_out appears to generate all the imported proto files too.

**Comment by ennerf** [comment]
I double checked and found an extra `file_to_generate` list that I wasn't aware of. That should be an easy fix.

```proto
message CodeGeneratorRequest {
  // The .proto files that were explicitly listed on the command-line.  The
  // code generator should generate code only for these files.  Each file's
  // descriptor will be included in proto_file, below.
  repeated string file_to_generate = 1;
  // ...
}
```

Source: [plugin.proto](https://github.com/protocolbuffers/protobuf/blob/main/src/google/protobuf/compiler/plugin.proto#L44-L71)

**Comment by PeterJohnson** [comment]
Any update on this?

**Comment by ennerf** [comment]
I was hoping to combine it with a PR for #59, but I haven't heard back yet

**Comment by ennerf** [comment]
Is this a blocking issue for you or can you work around it? I hope I can get to it soon.

**Comment by PeterJohnson** [comment]
We have a workaround that works for now.  Thanks.

**Comment by srimanachanta** [comment]
Any update on this?

**Comment by ennerf** [comment]
I just added the filter. I tested a few simple cases, but it'd be great if you could verify that this fixes your more complex cases too @PeterJohnson

**Comment by ennerf** [comment]
released in 1.3.3
